src/api/users/test/e2e/users.test.js

```diff
@@ -206,192 +85,101 @@ describe('PUT REQUESTS', () => {
-      .set('Content-Type', 'application/json')
-      .set('Authorization', `bearer ${createServiceToken()}`)
-      // But the newly updated data, with modified email
-      .send(user);
-
-    expect(response.statusCode).toBe(400);
-  });
-
-  test('Rejected - Update a nonexistent user', async () => {
-    const user = new User(defaultUsers.carl());
-
-    const response = await request(app)
-      .put(`/${user.id}`)
-      .set('Content-Type', 'application/json')
-      .set('Authorization', `bearer ${createServiceToken()}`)
-      .send(user);
-
-    expect(response.statusCode).toBe(404);
   });
-});
-
-describe('POST REQUESTS', () => {
-  beforeEach(clearData);
 
-  test('Accepted - Create a user', async () => {
-    const { response } = await createUser();
-    expect(response.statusCode).toBe(201);
-  });
+  it('should not post a user with missing first name info', async () => {
+    delete galileo.firstName;
+    const response = await postUser(galileo);
+    const parsedResponse = await response.json();
 
-  test('Rejected - Create two of the same user', async () => {
-    const { response: response1 } = await createUser();
-    expect(response1.statusCode).toBe(201);
+    expect(parsedResponse.statusCode).toBe(400);
+    expect(parsedResponse.validation.body.message).toBe('"firstName" is required');
 
-    const { response: response2 } = await createUser();
-    expect(response2.statusCode).toBe(400);
+    await deleteUser(galileo);
   });
 
-  test('Rejected - Ensure that the id param has to match the hashed email of the body', async () => {
-    const user1 = new User({ ...defaultUsers.carl() });
-    const user2 = new User({ ...defaultUsers.galileo() });
+  it('should not post a user with missing last name info', async () => {
+    delete galileo.lastName;
+    const response = await postUser(galileo);
+    const parsedResponse = await response.json();
 
-    const response = await request(app)
-      // Use user1's id
-      .post(`/${user1.id}`)
-      .set('Content-Type', 'application/json')
-      .set('Authorization', `bearer ${createServiceToken()}`)
-      // But user2's data
-      .send(user2);
+    expect(parsedResponse.statusCode).toBe(400);
+    expect(parsedResponse.validation.body.message).toBe('"lastName" is required');
 
-    expect(response.statusCode).toBe(400);
+    await deleteUser(galileo);
   });
 
-  test('Rejected - Ensure that the feeds array can only contain URI strings', async () => {
-    const user = new User({ ...defaultUsers.carl(), feeds: ['123'] });
+  it('should not allow the same user to be posted twice', async () => {
+    await postUser(galileo);
+    const response = await postUser(galileo);
+    const parsedResponse = await response.text();
 
-    const response = await request(app)
-      .post(`/${user.id}`)
-      .set('Content-Type', 'application/json')
-      .set('Authorization', `bearer ${createServiceToken()}`)
-      .send(user);
+    expect(response.status).toBe(400);
```

Comment:
Status code should be 409 instead of 400

Reply:
The returned code from `users` is 400 in this case.. Should we file a issue to change it to 409?

Reply:
Agreed, 409 is the correct code to be used here, so we should change it.

Reply:
Nothing stopping you from fixing that in this PR, since it means the test will pass.